### PR TITLE
Bug 1113281 - Update retrigger logic to always hit new retrigger endpoint

### PR DESCRIPTION
**Pull request by lightsofapollo**

### CODE REVIEW

File: webapp/app/js/models/job.js [hunk 1]
```diff
@@ -84,6 +84,18 @@ treeherder.factory('ThJobModel', [
         });
     };
 
+    ThJobModel.retrigger = function(repoName, pk, config) {
+        // a static method to retrieve a single instance of ThJobModel
```
Comment (camd) [inline webapp/app/js/models/job.js]: looks like this comment is incorrect

File: webapp/app/plugins/controller.js [hunk 2]
```diff
@@ -235,25 +231,43 @@ treeherder.controller('PluginCtrl', [
...
-                return null;
             }
         };
 
         $scope.retriggerJob = function() {
-            thBuildApi.retriggerJob($scope.repoName, getRequestId());
+            ThJobModel.retrigger($scope.repoName, $scope.job.id).then(function() {
+                var requestId = getBuildbotRequestId();
+                if (requestId) {
+                    return thBuildApi.retriggerJob($scope.repoName, requestId);
```
Comment (camd) [inline webapp/app/plugins/controller.js]: I'm not quite clear on why you call ``ThJobModel.retribber`` and then also call ``thBuildApi.retriggerJob``.  Wouldn't that do a retrigger for on buildbot jobs twice?

**Comment by lightsofapollo** [comment]
@camd After looking at this again I see I missed part of the cancel case fixing...

File: webapp/app/plugins/controller.js [hunk 3]
```diff
@@ -235,25 +231,49 @@ treeherder.controller('PluginCtrl', [
...
+              thNotify.send(message, "danger", true)
+            });
         };
 
         $scope.cancelJob = function() {
-            thBuildApi.cancelJob($scope.repoName, getRequestId()).then(function() {
-                ThJobModel.cancel($scope.repoName, $scope.job.id);
-            });
+            var requestId = getBuildbotRequestId();
+            if (requestId) {
```
Comment (camd) [inline webapp/app/plugins/controller.js]: This looks like it's the inverse of the retrigger.  Should we call the ``ThJobModel.cancel`` first and then the ``thBuildApi.cancelJob`` in the ``then`` section if we find the ``requestId``?

File: webapp/app/plugins/controller.js [hunk 4]
```diff
@@ -235,25 +231,49 @@ treeherder.controller('PluginCtrl', [
...
+                  ThJobModel.cancel($scope.repoName, $scope.job.id);
+              });
+            } else {
+              thNotify.send(
+                "Unable to get request id for retrigger/cancel", "danger", true
+              );
+            }
         };
 
         $scope.cancelAll = function(resultsetId) {
```
Comment (camd) [inline webapp/app/plugins/controller.js]: is there any concept of the pulse cancel for a group of jobs like this for the result set?

Comment (lightsofapollo) [inline webapp/app/plugins/controller.js]: No one job = one cancel (cancel all handles this as sending a one message per job that was cancelled on the server side)

File: webapp/app/plugins/controller.js (hunk 3 above)
Comment (lightsofapollo) [inline webapp/app/plugins/controller.js]: updated now sorry for confusion here

**Comment by camd** [comment]
@lightsofapollo - This looks good to me.  I take it you tested it and it works ok?  :)  
Are you able to merge this yourself?  Or should I merge it?

**Comment by lightsofapollo** [comment]
@camd Going to give it another check and then merge

**Comment by edmorley** [comment]
I'm wondering if we should hold off merging any more changes greater than a few lines until we get this deploy working. We're already starting to get pile-on effect with later additions causing more regressions.

**Pull request merged**